### ISSUE TITLE
Layered Histogram fails to represent null values

### ISSUE THREAD

**Issue by harabat** [altair-viz/altair#2411]
## Issue

The method for making layered histograms [suggested on altair-viz.github.io](https://altair-viz.github.io/gallery/layered_histogram.html) seems to fail to take into account null values within the range of the data: bins that should be empty are represented as having 1 observation. 

---

## Examples

Altair:
```python
import pandas as pd
import altair as alt
import numpy as np

np.random.seed(42)

# Generating Data
source = pd.DataFrame({
    'Trial A': np.random.normal(0, 0.8, 1000),
    'Trial B': np.random.normal(-2, 1, 1000),
    'Trial C': np.random.normal(3, 2, 1000)
})

alt.Chart(source).transform_fold(
    ['Trial A', 'Trial B', 'Trial C'],
    as_=['Experiment', 'Measurement']
).mark_area(
    opacity=0.3,
    interpolate='step'
).encode(
    alt.X('Measurement:Q', bin=alt.Bin(maxbins=100)),
    alt.Y('count()', stack=None),
    alt.Color('Experiment:N')
)
```
![layered_histogram_altair](https://user-images.githubusercontent.com/45575009/108603170-45eb2f00-739e-11eb-88fe-0b1468d33a1c.png)
---

Seaborn:
```python
import seaborn as sns

source_melt = source.melt(var_name='Experiment', value_name='Measurement')
sns.displot(
    source_melt,
    x='Measurement',
    binwidth=.2,
    hue='Experiment',
    aspect=2,
    element='step'
)
```
![layered_histogram_seaborn](https://user-images.githubusercontent.com/45575009/108603174-4b487980-739e-11eb-9db2-c52d958f6297.png)

---

## Solutions

The easiest solution would be to suggest another method. The one below solves the null values in bins, but is much more verbose, and requires manually setting the x-axis and the legend. 

```python
trial_a = alt.Chart(source).mark_bar(
    opacity=0.5, color='#c9d6e5'
).encode(
    alt.X('Trial A:Q', bin=alt.Bin(maxbins=50), axis=None),
    y='count()'
)

trial_b = alt.Chart(source).mark_bar(
    opacity=0.5, color='#fcdab9'
).encode(
    alt.X('Trial B:Q', bin=alt.Bin(maxbins=50), axis=None),
    y='count()'
)

trial_c = alt.Chart(source).mark_bar(
    opacity=0.5, color='#f7cccc'
).encode(
    alt.X('Trial C:Q', bin=alt.Bin(maxbins=100), axis=None),
    y='count()'
)

trial_a + trial_b + trial_c
```
![layered_histogram_altair_solution](https://user-images.githubusercontent.com/45575009/108603706-6d8fc680-73a1-11eb-9222-b1f4a707174e.png)

Another would be to fix the issue. If this is a vega-lite issue, I can move it there, but thought I'd get an opinion here first.

**Comment by harabat** [comment]
Solved it by replacing `mark_area` by `mark_bar`. I'll make a PR to the altair-viz.github.io repo. 

Closing the issue.

**Comment by jakevdp** [comment]
Reopening because this is the appropriate place for the issue (the website repo contains automatically-generated outputs of code contained in this repo).

The relevant Vega-Lite issue is this: https://github.com/vega/vega-lite/issues/5403 There is no good solution that I know of.

**Comment by harabat** [comment]
Hey @jakevdp, should I make a change to the file [altair/examples/layered_histogram.py](https://github.com/altair-viz/altair/blob/master/altair/examples/layered_histogram.py) with my solution (replacing `mark_area` by `mark_bar` and removing `interpolate='step'`) in the meantime?

I'd also like to thank you for all your hard work! `Altair` is amazing, and I always see you making in-depth answers to questions about `Altair` on stackoverflow and HN.

**Comment by joelostblom** [comment]
@harabat Yup that is the right place. After you have made the changes you can build the documentation locally to make sure that it shows up as you indended with `make html` from the `doc` dir. To make the histogram look like an area without the whitelines between the bars, you can set the `stroke` encoding to the same as the `color`. I have seen a couple of other issues about the step-interpolation not working exactly as a histogram, so I think you can make your suggested change to this example as well https://altair-viz.github.io/gallery/scatter_marginal_hist.html

**Comment by joelostblom** [comment]
This was resolved in #2419
